api/rest/bundle_handler.go

```diff
@@ -235,6 +235,18 @@ func (h BundleHandler) Get(w http.ResponseWriter, r *http.Request) {
 func (h BundleHandler) GetFile(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	id := vars["id"]
+
+	bundle, err := h.getBundleState(id)
+	if err != nil {
+		writeJSONError(w, http.StatusInternalServerError, err)
+		return
+	}
+	if bundle.Status != Done {
+		writeJSONError(w, http.StatusNotFound,
```

Comment:
If we end up with a bundle in the `Failed` state, should that change our error? Is it even possible to get that with cluster bundles?

Reply:
Good point! If bundle is Canceled, Failed or Deleted we should return 404

Reply:
Failed state is here https://github.com/dcos/dcos-diagnostics/pull/150

Reply:
Fixed